File: app/src/main/java/jimmysharp/kanaclogger/MainActivity.java
package jimmysharp.kanaclogger;

import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import com.squareup.sqlbrite.BriteDatabase;
import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    DatabaseInitializer dbmanager = null;
    BriteDatabase db = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dbmanager = new DatabaseInitializer(this.getApplicationContext());

        try {
            db = dbmanager.open();
        } catch (IOException e) {
            this.finish();
        }

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_main);
        setSupportActionBar(toolbar);

        FragmentManager manager = getFragmentManager();
        FragmentTransaction transaction = manager.beginTransaction();
        transaction.replace(R.id.container_main, new ContentsFragment());
        transaction.commit();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        dbmanager.close();
        dbmanager.dispose();
        dbmanager = null;
    }

    public BriteDatabase getDB(){
        return db;
    }
}

File: app/src/main/java/jimmysharp/kanaclogger/app/main/MainActivity.kt
package jimmysharp.kanaclogger.app.main

import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.view.Menu
import android.view.MenuItem

import jimmysharp.kanaclogger.R
import jimmysharp.kanaclogger.app.settings.SettingsActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    companion object {
        val TAG = MainActivity::class.java.simpleName
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(toolbar_main)

        val pager = SectionsPagerAdapter(supportFragmentManager)

        container_main.adapter = pager
        container_main.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabs_main))
        tabs_main.addOnTabSelectedListener(TabLayout.ViewPagerOnTabSelectedListener(container_main))
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id == R.id.action_settings) {
            val intent = Intent(this@MainActivity, SettingsActivity::class.java)
            startActivity(intent)
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    inner class SectionsPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment? {
            var fragment: Fragment? = null
            when (position) {
                0 -> fragment = BulkRegisterFragment.newInstance()
                1 -> fragment = ConstructionsFragment.newInstance()
                2 -> fragment = DropsFragment.newInstance()
                3 -> fragment = ShipListFragment.newInstance()
            }
            return fragment
        }

        override fun getCount(): Int {
            return 4
        }
    }
}
